File: 0000-0999/0068.TextJustification.go
func fullJustify(words []string, maxWidth int) []string {
    ans := []string{}
    line, width := []string{}, 0
    for _, word := range words {
        n := len(line)
        if width + len(word) + n <= maxWidth {
            line = append(line, word)
            width += len(word)
        } else {
            if n == 1 {
                ans = append(ans, line[0]+strings.Repeat(" ", maxWidth-width))
            } else {
                space, extra := (maxWidth-width)/(n-1), (maxWidth-width)%(n-1)
                for i := 0; i < extra; i++ {
                    line[i] += " "
                }
                ans = append(ans, strings.Join(line, strings.Repeat(" ", space)))
            }
            line, width = []string{word}, len(word)
        }
    }
    last := strings.Join(line, " ")
    ans = append(ans, last+strings.Repeat(" ", maxWidth-len(last)))
    return ans
}

File: 0000-0999/0068.TextJustification.py
class Solution(object):
    def fullJustify(self, words, maxWidth):
        """
        :type words: List[str]
        :type maxWidth: int
        :rtype: List[str]
        """
        ans = []
        cur, temp = [], 0
        for word in words:
            if temp + len(cur) + len(word) > maxWidth:
                for i in xrange(maxWidth-temp):
                    cur[i%(len(cur)-1 or 1)] += ' '
                ans.append(''.join(cur))
                cur, temp = [], 0
            cur.append(word)
            temp += len(word)
        return ans+[' '.join(cur).ljust(maxWidth)]
